### ISSUE TITLE
EarlyStopping is returning null in getBestModel 

### ISSUE THREAD

**Issue by MatheusCalil** [deeplearning4j/deeplearning4j#4054]
#### Issue Description

Please describe our issue, along with:
- expected behavior
The best model with my termination conditions (https://deeplearning4j.org/earlystopping)
- encountered behavior
getBestModel() is returning null

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version
0.9.2-SNAPSHOT
* platform information (OS, etc)
Windows 10

My Code:

` 

        public static MultiLayerNetwork earlyStopLeNet(DataSetIterator train, DataSetIterator test, int width, int height, int seed, int iterations){

			ZooModel zooModel = new LeNet (2, seed, iterations);
			int[][] shape = {{1, width, height}};
			zooModel.setInputShape(shape);
			MultiLayerNetwork  net = (MultiLayerNetwork) zooModel.init();

			EarlyStoppingModelSaver<MultiLayerNetwork> saver = new InMemoryModelSaver();
			EarlyStoppingConfiguration<MultiLayerNetwork> esConf = new EarlyStoppingConfiguration.Builder()
					//.epochTerminationConditions(new MaxEpochsTerminationCondition(30))
					.iterationTerminationConditions(new MaxTimeIterationTerminationCondition(20, TimeUnit.MINUTES))
					.scoreCalculator(new DataSetLossCalculator(test, true))
			        .evaluateEveryNEpochs(1)
					.modelSaver(saver)
					.build();
	
			EarlyStoppingTrainer trainer = new EarlyStoppingTrainer(esConf,net,train);
	
			//Conduct early stopping training:
			EarlyStoppingResult result = trainer.fit();
	
			//Print out the results:
			System.out.println("Termination reason: " + result.getTerminationReason());
			System.out.println("Termination details: " + result.getTerminationDetails());
			System.out.println("Total epochs: " + result.getTotalEpochs());
			System.out.println("Best epoch number: " + result.getBestModelEpoch());
			System.out.println("Score at best epoch: " + result.getBestModelScore());
	
			//Get the best model:
			MultiLayerNetwork bestModel = (MultiLayerNetwork) result.getBestModel();

			return bestModel;
			
		}
`

Where DataSetIterator train and DataSetIterator test are RecordReaderDataSetIterator that receive a ImageRecordReader as parameter.
This model (LeNet from ZooModel) works fine with this parameters when I don't try to use EarlyStopping.

The output in system.out is giving:
Termination reason: IterationTerminationCondition
Termination details: MaxTimeIterationTerminationCondition(20,unit=MINUTES)
Total epochs: 0
Best epoch number: -1
Score at best epoch: 1.7976931348623157E308

Any ideas that why this could be happening?

**Comment by madppiper** [comment]
I have the exact problem with precisely the same values:

2017-11-10 16:48:11 16:48:11,785 INFO  Termination reason: EpochTerminationCondition
2017-11-10 16:48:11 16:48:11,785 INFO  Termination details: MaxEpochsTerminationCondition(400)
2017-11-10 16:48:11 16:48:11,785 INFO  Total epochs: 400
2017-11-10 16:48:11 16:48:11,785 INFO  Best epoch number: -1
2017-11-10 16:48:11 16:48:11,785 INFO  Score at best epoch: 1.7976931348623157E308

Did anybody ever find a solution for this?
